src/include/dhandle.h

```diff
@@ -72,6 +72,13 @@
         }                                                                 \
     } while (0)
 
+enum wt_dhandle_type {
+    WT_DHANDLE_TYPE_BTREE,
+    WT_DHANDLE_TYPE_TABLE,
+    WT_DHANDLE_TYPE_TIERED,
+    WT_DHANDLE_TYPE_TIERED_TREE
+};
```

Comment:
Should we add an extra "max" value?
```suggestion
enum wt_dhandle_type {
    WT_DHANDLE_TYPE_BTREE,
    WT_DHANDLE_TYPE_TABLE,
    WT_DHANDLE_TYPE_TIERED,
    WT_DHANDLE_TYPE_TIERED_TREE,
    WT_DHANDLE_TYPE_MAX
};
```

This way we can use `WT_DHANDLE_TYPE_MAX` wherever convenient: in the WT_ASSERT, array size, ...

Reply:
If you write a switch on an enum without a default: the compiler gives an error if all values of the enum are not accounted for. If you introduce a max value that doesn't work unless the max value is a macro instead of an enum value.

So I introduced WT_DHANDLE_TYPE_MAX as a macro.

P.S. The reason I moved enum here from where it was is a compiler error when left where it was.

Reply:
The idea behind having a max invalid value is to avoid `1 + WT_DHANDLE_TYPE_TIERED_TREE` as it can become wrong overtime.

If we add a new dhandle type, let's say `WT_DHANDLE_TYPE_TIERED_TREE_V2`, after `WT_DHANDLE_TYPE_TIERED_TREE`, the definition of `1 + WT_DHANDLE_TYPE_TIERED_TREE` becomes invalid.

Using `WT_DHANDLE_TYPE_MAX` would always remain valid even if we add new dhandle types, as long as they are defined before `WT_DHANDLE_TYPE_MAX`.